mod/htmlawed/start.php

```diff
@@ -14,7 +14,13 @@
  * Initialize the htmlawed plugin
  */
 function htmlawed_init() {
-	elgg_register_plugin_hook_handler('validate', 'input', 'htmlawed_filter_tags', 1);
+
+	// just to show off the factory feature
+	elgg_register_class_factory(\HtmlawedPlugin\ValidateInputHook::KLASS, function ($class) {
```

Comment:
This PR seems to mix DI with the class-as-handler feature. Can we just limit it to class-as-handler and do a "proper" DI implementation later?